common/ext/class.Namespace.php

```diff
@@ -18,148 +18,100 @@
-
+    protected $uri;
+
     /**
-     * Create a namespace instance
+     * Namespace constructor.
      *
-     * @access public
      * @author Jerome Bogaerts, <jerome.bogaerts@tudor.lu>
-     * @param  int id
+     *
+     * @param  string id
      * @param  string uri
-     * @return mixed
      */
-    public function __construct($id = 0, $uri = '')
+    public function __construct($id = '', $uri = '')
     {
-        
-        
-        if ($id > 0) {
-            $this->modelId = $id;
-        }
-        if (!empty($uri)) {
-            $this->uri = $uri;
-        }
+        $this->modelId = $id;
+        $this->uri = $uri;
     }
-
+
     /**
      * Get the identifier of the namespace instance
      *
-     * @access public
      * @author Jerome Bogaerts, <jerome.bogaerts@tudor.lu>
-     * @return int
+     * @return string
      */
     public function getModelId()
     {
-        $returnValue = (int) 0;
-
-        
-        
-        $returnValue = $this->modelId;
-        
-        
-
-        return (int) $returnValue;
+        return $this->modelId;
     }
-
+
     /**
      * Get the namespace URI
      *
-     * @access public
      * @author Jerome Bogaerts, <jerome.bogaerts@tudor.lu>
      * @return string
      */
     public function getUri()
     {
-        $returnValue = (string) '';
-
-        
-        
-        $returnValue = $this->uri;
-        
-        
-
-        return (string) $returnValue;
+        return $this->uri;
     }
-
+
     /**
      * Magic method, return the Namespace URI
      *
-     * @access public
      * @author Jerome Bogaerts, <jerome.bogaerts@tudor.lu>
      * @return string
      */
     public function __toString()
     {
-        $returnValue = (string) '';
-
-        
-        
-        $returnValue = $this->getUri();
-        
-        
-
-        return (string) $returnValue;
+        return $this->getUri();
     }
-
+
     /**
      * Remove a namespace from the ontology. All triples bound to the model will
      * be removed.
      *
-     * @access public
      * @author Jerome Bogaerts, <jerome.bogaerts@tudor.lu>
+     *
      * @return boolean
+     * @throws core_kernel_persistence_Exception
      */
     public function remove()
     {
-        $returnValue = (bool) false;
-
-        
         $db = core_kernel_classes_DbWrapper::singleton();
+
+        // TODO refactor this to use triple store abstraction.
         if (false === $db->exec("DELETE FROM statements WHERE modelid = ?", [$this->getModelId()])) {
-            $returnValue = false;
-        } else {
-            if (false === $db->exec("DELETE FROM models WHERE modelid = ?", [$this->getModelId()])) {
-                $returnValue = false;
-            } else {
-                $returnValue = true;
-            }
+            return false;
+        }
+
+        // TODO refactor this to use triple store abstraction.
+        if (false === $db->exec("DELETE FROM models WHERE modelid = ?", [$this->getModelId()])) {
```

Comment:
I think here you can return directly the `$db->exec("DELETE FROM models WHERE modelid = ?", [$this->getModelId()])`
the issue i see is that from 2 times trying to delete the mode we are doing it now only once,
so please check this that we are not changing the scope of this code block

Reply:
the behaviour is the same as before. I have factorize the code to return directly exec ouput

Reply:
No, this first removes every **statement** linked to a model id, then the **model** itself.